src/sphinx_argparse_cli/_logic.py

```diff
@@ -126,8 +130,35 @@ def run(self) -> list[Node]:
         return [home_section]
 
     def _mk_option_group(self, group: _ArgumentGroup, prefix: str) -> section:
-        title_text = f"{prefix}{' ' if prefix else ''}{group.title}"
-        ref_id = make_id(title_text)
+        sub_title_prefix: str = self.options["group_sub_title_prefix"]
```

Comment:
This change and the other below seems to share a lot of common code, can we extract the commonality into a method? Thanks!

Reply:
Are you sure? I think the resemblance is only apparent. If you take a look carefully you will see that:

- In `_mk_option_group` is checking if `" "` exists in prefix because groups can render groups inside subcommands (for example, optional arguments), but subcommands can't have `" "` in their prefixes (`<prog>` comes from subparsers).
- Prefix splitting for `None` cases in options is totally different.

I've tried to write a common method at first, but it needed a lot of parameters, so many that they were obfuscating the logic rather than clarifying the behavior, and some conditional branches couldn't be removed at all.

Reply:
Yeah, understand but still seems fairly close. I can take a dig at it myself later.